core/cas-server-core-webflow-mfa-api/src/main/java/org/apereo/cas/web/flow/configurer/AbstractCasMultifactorWebflowConfigurer.java

```diff
@@ -90,6 +99,40 @@ protected void augmentMultifactorProviderFlowRegistry(final FlowDefinitionRegist
      * @param mfaProviderFlowRegistry the registry
      */
     protected void registerMultifactorProviderAuthenticationWebflow(final Flow flow, final String subflowId, final FlowDefinitionRegistry mfaProviderFlowRegistry) {
+        val mfaFlow = (Flow) mfaProviderFlowRegistry.getFlowDefinition(subflowId);
+
+        // Insert bypass, available and failure actions into the flow.
+        val initLoginState = getState(mfaFlow, CasWebflowConstants.STATE_ID_INIT_LOGIN_FORM, ActionState.class);
```

Comment:
Is this going to work, if the webflow transition never comes to init-login-form, such as the case with delegated authn?

Reply:
Yes, the delegated authn flow configurer and all the mfa subflow configurers as far as I know start by calling getLoginFlow(), and use the webflow.xml as a starting template.  If that template changes or someone creates a flow that is not built from that template there could be issues.

Reply:
They all do operate on the login flow, sure, but they dont all come to the same state in that flow. For example, in delegated authn flow there is no need to "init the login form" because there is no login form. The authn is handed off to an external identity provider that has the login form, but on the return trip, MFA should trigger by CAS. (Same would be true for X509, etc) Given the way you are modifying the webflow here, I suspect this may break those integrations.

Have you tested a scenario where you are handed off to Twitter for instance for authn, and on the return trip, CAS would do duo?